src/modules/endpoint/reducers/morphContentLanguage.js

```diff
@@ -1,83 +1,83 @@
-import { reduce } from 'lodash/collection'
-import { CategoryModel } from '@integreat-app/integreat-api-client'
+import { EventModel } from '@integreat-app/integreat-api-client'
 import type { MorphContentLanguageActionType } from '../../app/StoreActionType'
-import type {
-  CategoryRouteStateType,
-  CityContentStateType,
-  EventRouteStateType
-} from '../../app/StateType'
-
-const translatePath = (model: CategoryModel, currentCity: string, newLanguage: string) => {
-  if (model.isRoot()) {
-    return `/${currentCity}/${newLanguage}`
-  } else {
-    return model.availableLanguages.get(newLanguage)
-  }
-}
-
-const translateChildren = (models, newCategoriesMap, children, currentCity, newLanguage) =>
-  reduce(children, (result, value: Array<string>, path: string) => {
-    const translatedKey = translatePath(models[path], currentCity, newLanguage)
+import CategoriesMapModel from '@integreat-app/integreat-api-client/models/CategoriesMapModel'
+import forEachTreeNode from '../../common/forEachTreeNode'
+import { isEmpty } from 'lodash'
 
-    if (!translatedKey) {
-      // TODO: This is code for debugging which could help in the future. Remove once this has been tested in NATIVE-116
-      console.warn(`Path ${path} is not translatable!`)
-      return result
-    }
+const categoryRouteTranslator = (newCategoriesMap: CategoriesMapModel, city: string, newLanguage: string) =>
+  (route: CategoryRouteStateType, navKey: string): CategoryRouteStateType => {
+    try {
+      const {depth, root, allAvailableLanguages} = route
 
-    if (!newCategoriesMap.findCategoryByPath(translatedKey)) {
-      // TODO: This is code for debugging which could help in the future. Remove once this has been tested in NATIVE-116
-      console.warn(`Path ${translatedKey} does not exist in new model!`)
-      return result
-    }
+      const translatedRoot = allAvailableLanguages.get(newLanguage)
 
-    const translatedArray = reduce(children[path], (result, key) => {
-      const translatedKey = models[key].availableLanguages.get(newLanguage)
-      if (!translatedKey) {
-        // TODO: This is code for debugging which could help in the future. Remove once this has been tested in NATIVE-116
-        console.warn(`Path ${key} is not translatable!`)
-        return result
+      if (!translatedRoot) { // Route is not translatable
+        return {...route}
       }
 
-      if (!newCategoriesMap.findCategoryByPath(translatedKey)) {
-        // TODO: This is code for debugging which could help in the future. Remove once this has been tested in NATIVE-116
-        console.warn(`Path ${translatedKey} does not exist in new model!`)
-        return result
+      const rootModel = newCategoriesMap.findCategoryByPath(translatedRoot)
+      if (!rootModel) {
+        console.warn(`Inconsistent data detected: ${translatedRoot} does not exist,
+                      but is referenced in ${root} as translation for ${newLanguage}.`)
+        return {...route}
```

Comment:
Same again.

Reply:
Done.

Reply:
There were some more `{...route}` did you also fix  those?